### PR TITLE
Dont't include ignored packages when adding a changeset

### LINKED ISSUE #436: Choosing all packages to bump also includes ignored packages and results in error during version phase
## Affected Packages

- not sure, guess cli

## Problem

I tried to create a changeset using `yarn chageset`. I needed to bump all packages in your monorepo. So I choose `all changes package` and `all unchanged packages`. It results in adding ignored packages (defined in `ignore` in the config files) to the list of changes. 

When trying to run `yarn changesets version` command, it now thrown an errors:

```
Mixed changesets that contain both ignored and not ignored packages are not allowed
```

## Proposed solution

Remove ignored packages from the selection list in `yarn changeset` command? 🤔

### PR DESCRIPTION
**Pull request by mskelton**
Fixes #436

Filters ignored packages before displaying the list of packages when adding a new changeset.

### CODE REVIEW

**Comment by jsheely** [comment]
I'd like to see this get merged in soon. Having to skip over sample apps in our mono repo every time we publish is becoming a chore

**Comment by mskelton** [comment]
@Andarist Any chance of getting this merged in?

**Comment by StarpTech** [comment]
Also a ping from me. Please merge it.

**Comment by wycats** [comment]
I would also find this extremely useful and consider it a blocker for fully adopting changesets in [@starbeamjs/starbeam](https://github.com/starbeamjs/starbeam)

**Comment by dotansimha** [comment]
Same here. Anything that can be done to get this merged? :) @Andarist thanks

**Comment by silesky** [comment]
@Andarist is there something I can do to help merge this?

**Comment by wycats** [comment]
At this point, would it be appropriate to temporarily fork and publish this with the changes?

File: packages/cli/src/commands/add/index.ts [hunk 1]
```diff
@@ -22,7 +22,9 @@ export default async function add(
   { empty, open }: { empty?: boolean; open?: boolean },
   config: Config
 ) {
-  const packages = await getPackages(cwd);
+  const packages = (await getPackages(cwd)).packages.filter(
```
[the remark this answers is not in the document]
Comment (mitchellhamilton) [reply packages/cli/src/commands/add/index.ts:26]: The message added when `repoHasMultiplePackages` is true is about bumping dependents which if there is only one non-ignored package is irrelevant so I'd say the behavior in this PR is correct.

**Review by mitchellhamilton — APPROVED** [review]

**Comment by wycats** [comment]
> At this point, would it be appropriate to temporarily fork and publish this with the changes?

This is one of happiest thumbs-downs I've ever received on Github 😄

File: packages/cli/src/commands/add/index.ts [hunk 2]
```diff
@@ -39,10 +41,11 @@ export default async function add(
       ref: config.baseBranch
     });
     const changePackagesName = changedPackages
-      .filter(a => a)
-      .map(pkg => pkg.packageJson.name);
-    newChangeset = await createChangeset(changePackagesName, packages.packages);
-    printConfirmationMessage(newChangeset, packages.packages.length > 1);
+      .map(pkg => pkg.packageJson.name)
+      .filter(pkgName => config.ignore.includes(pkgName));
```
Comment (glasser) [inline packages/cli/src/commands/add/index.ts:45]: Is there a missing `!` here?

**Comment by mskelton** [comment]
@wycats https://github.com/changesets/changesets/pull/871 was already opened to fix the bug.  Somehow I missed the not in my original PR, but there is already a PR to fix it.

**Comment by mskelton** [comment]
@Andarist I just verified the fix applied in 2.23.2 works correctly.  Thanks for the quick resolution to my mistake!

**Comment by wycats** [comment]
@mskelton I apologize if my tone sounds aggressive or annoying. I'm just really excited about the prospect of building changesets into the [Starbeam](https://github.com/starbeamjs) release workflow, so I'm eager (overeager?) to help 😳.

I already did a handful of releases using Changesets, and I'm cautiously optimistic that with this bugfix (and a few minor tweaks to how we're using it now, including figuring out how to get the exact automation we need for PRs) that we'll be using it for years to come.

Thanks so much for your engagement 🙌

**Pull request merged**